jena-text/src/main/java/org/apache/jena/query/text/TextIndexES.java

```diff
@@ -0,0 +1,427 @@
+        if(!exists.isExists()) {
+            Settings indexSettings = Settings.builder()
+                    .put(NUM_OF_SHARDS, esSettings.getShards())
+                    .put(NUM_OF_REPLICAS, esSettings.getReplicas())
+                    .build();
+            LOGGER.debug("Index with name " + INDEX_NAME + " does not exist yet. Creating one with settings: " + indexSettings.toString());
+            client.admin().indices().prepareCreate(INDEX_NAME).setSettings(indexSettings).get();
+        }
+
+
+
+    }
+
+
+    /**
+     * Constructor used mainly for performing Integration tests
+     * @param config an instance of {@link TextIndexConfig}
+     * @param client an instance of {@link TransportClient}. The client should already have been initialized with an index
+     */
+    public TextIndexES(TextIndexConfig config, Client client, String indexName) {
+        this.docDef = config.getEntDef();
+        this.isMultilingual = true;
+        this.client = client;
+        this.INDEX_NAME = indexName;
+    }
+
+    /**
+     * We do not have any specific logic to perform before committing
+     */
+    @Override
+    public void prepareCommit() {
+        //Do Nothing
+
+    }
+
+    /**
+     * Commit happens in the individual get/add/delete operations
+     */
+    @Override
+    public void commit() {
+        // Do Nothing
+    }
+
+    /**
+     * not really sure what we need to roll back.
+     */
+    @Override
+    public void rollback() {
+       //Not sure what to do here
+
+    }
+
+    /**
+     * We don't have resources that need to be closed explicitely
+     */
+    @Override
+    public void close() {
+        // Do Nothing
+
+    }
+
+    /**
+     * Update an Entity. Since we are doing Upserts in add entity anyways, we simply call {@link #addEntity(Entity)}
+     * method that takes care of updating the Entity as well.
+     * @param entity the entity to update.
+     */
+    @Override
+    public void updateEntity(Entity entity) {
+        //Since Add entity also updates the indexed document in case it already exists,
+        // we can simply call the addEntity from here.
+        addEntity(entity);
+    }
+
+
+    /**
+     * Add an Entity to the ElasticSearch Index.
+     * The entity will be added as a new document in ES, if it does not already exists.
+     * If the Entity exists, then the entity will simply be updated.
+     * The entity will never be replaced.
+     * @param entity the entity to add
+     */
+    @Override
+    public void addEntity(Entity entity) {
+        LOGGER.debug("Adding/Updating the entity in ES");
+
+        //The field that has a not null value in the current Entity instance.
+        //Required, mainly for building a script for the update command.
+        String fieldToAdd = null;
+        String fieldValueToAdd = "";
+        try {
+            XContentBuilder builder = jsonBuilder()
+                    .startObject();
+
+            //Currently ignoring Graph field based indexing
```

Comment:
If you don't intend to support it (and I believe it would take a lot of effort, not just this code snippet), I suggest you remove the currently commented out code.